File: leetcode/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        for index, element in enumerate(nums):
            complement = target - element
            if complement in num_map:
                return [num_map[complement], index]
            num_map[element] = index
        return []

File: leetcode/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> num_map = new HashMap<>();
        int n = nums.length;

        for(int i = 0; i < n; i++){
            int complement = target - nums[i];
            if (num_map.containsKey(complement)){
                return new int[]{num_map.get(complement), i};
            }
            num_map.put(nums[i], i);
        }
        return new int[]{};
    }
}
